rcode-reader/src/main/java/eu/rssw/pct/RCodeInfo.java

```diff
@@ -0,0 +1,723 @@
+      fn.returnType = DataType.getDataType(retType.substring(0, retType.lastIndexOf(' ')));
+      if (fn.returnType == DataType.CLASS) {
+        fn.classReturnType = retType.substring(0, retType.indexOf(' '));
+      }
+    }
+    
+    // Parameters
+    while (lst.hasNext()) {
+      String str2 = lst.next();
+      if (!str2.isEmpty()) {
+        // Generics handling, which break comma-separated list...
+        if (str2.contains("[[")) {
+          while (!str2.contains("]]")) {
+            str2 = str2 + lst.next();
+          }
+        }
+
+        fn.parameters.add(parseParameter(str2));
+      }
+    }
+    
+    return fn;
+  }
+
+  private static Parameter parseParameter(String str) {
+    Parameter param = new Parameter();
+    List<String> prm = Splitter.on(' ').trimResults().splitToList(str);
+    if (prm.size() != 4) {
+      return param;
+    }
+    param.name = prm.get(1);
+    param.type = ParameterType.getTypeFromString(prm.get(0));
+    param.datatype = DataType.getDataType(prm.get(2));
+    param.extent = Integer.parseInt(prm.get(3));
+    param.classType = prm.get(2);
+
+    return param;
+  }
+
+  // Main object for all object signatures
+  public static class RCodeUnit {
+    private Function mainProcedure;
+    private Collection<Function> funcs = new ArrayList<>();
+    private List<TempTable> tts = new ArrayList<>();
+    private Collection<MethodElement> methods = new ArrayList<>();
+    private Collection<PropertyElement> properties = new ArrayList<>();
+    private Collection<EventElement> events = new ArrayList<>();
+    private Collection<VariableElement> variables = new ArrayList<>();
+    private Collection<TableElement> tables = new ArrayList<>();
+    private Collection<BufferElement> buffers = new ArrayList<>();
+
+    public Function getMainProcedure() {
+      return mainProcedure;
+    }
+
+    public Collection<Function> getFuncs() {
+      return funcs;
+    }
+
+    public Collection<MethodElement> getMethods() {
+      return methods;
+    }
+
+    public Collection<PropertyElement> getProperties() {
+      return properties;
+    }
+
+    public Collection<EventElement> getEvents() {
+      return events;
+    }
+
+    public Collection<VariableElement> getVariables() {
+      return variables;
+    }
+
+    public Collection<TableElement> getTables() {
+      return tables;
+    }
+
+    public Collection<BufferElement> getBuffers() {
+      return buffers;
+    }
+  }
+  
+  public static class Function {
+    private EnumSet<SigAccessType> accessTypes;
+    private String name;
+    private List<Parameter> parameters = new ArrayList<>();
+    private DataType returnType;
+    private String classReturnType;
+    private int returnExtent;
+    private FunctionType type;
+
+    @Override
+    public String toString() {
+      StringBuilder sb = new StringBuilder();
+      for (SigAccessType type : accessTypes) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "type" which hides the field declared at line 570. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AHiddenFieldCheck)